[kubernetes] updates and new plans

schu opened the pull request:
This pull request updates the existing Kubernetes plan to latest stable (1.8.3 at the time of writing). It also adds plans for all components required for a Kubernetes cluster setup with Habitat.

Right now the components (i.e. `kubernetes-*` packages) start from version `0.1.0`. I'm undecided if they should rather match the Kubernetes version, i.e. here be `1.8.3` or `1.8.3-1` instead. Thoughts?

We will share an example "Kubernetes The Hab Way" setup based on this packages soon (probably later this or next week).

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

rsertelon commented:
@schu PR names in core-plans should start with `[packagename]` :) could you please rename it? :)

schu commented:
@rsertelon thanks, missed that, done.

rsertelon commented:
@schu np, I removed my comment, because I thought this was actually not a big deal `Kubernetes:` vs `[kubernetes]` :) Thanks for doing it :)

bdangit commented:
This is fantastic!  There is one concern which is if we are using the prebuilt binaries, they may not be linked to Hab’s glibc.  Unless the packages already can work across any Linux OS.

On versioning, Hab doesn’t really do semenatic versioning and it’s pretty bad to lock the version.  I would say leave your wrappers at 0.1.0.  Versioning them with k8s doesn’t make much sense since we aren’t locking the version of k8s dependency.  If you ever had to lock your wrappers to a specific version, then we should embed some of version into the pkg_name. Then it would be possible to lock a set of wrappers to a specific version of your dependency.

schu commented:
Updated.

> There is one concern which is if we are using the prebuilt binaries, they may not be linked to Hab’s glibc. Unless the packages already can work across any Linux OS.

I dropped the part using prebuilt upstream binaries and only updated to latest k8s stable (see https://github.com/habitat-sh/core-plans/pull/967/commits/c495c6311fa55c9298483c93adad2fe0912a1663).

> On versioning, Hab doesn’t really do semenatic versioning and it’s pretty bad to lock the version. I would say leave your wrappers at 0.1.0. Versioning them with k8s doesn’t make much sense since we aren’t locking the version of k8s dependency. If you ever had to lock your wrappers to a specific version, then we should embed some of version into the pkg_name. Then it would be possible to lock a set of wrappers to a specific version of your dependency.

The version actually was locked previously, I removed that now. Going with `0.1.0` for the time being. Can be revisited later, if necessary.

Thank you for the review! :)

schu commented:
Updated.

> We need a couple more things though which is - if you could add a README.md to core/kubernetes that explains how each these work together and how to start them. For examples of such readmes take a look at core/postgresql or core/concourse.

Done with https://github.com/habitat-sh/core-plans/pull/967/commits/40a505bb2ff2222fcff23c220289acdda47b8100 I feel detailed instructions are out of scope as this would mean to maintain a Habitat-specific Kubernetes setup manual. I hope referring to KTHW / the demo instead is ok.?

> The last bit is you will need to modify the builder.toml so that each of these reverse dependent packages will be rebuilt when core/kubernetes is rebuilt.

Done with https://github.com/habitat-sh/core-plans/pull/967/commits/192bb6e42f5517a28b1c788b06ea8809ebeaa66e

eeyun commented:
For the sake of dialogue I think it's important to note that adding detailed instructions like you have in KTHW is definitely not out of scope for the repo. As we get more high level service packages we will be in need of readme's like the one you provided there.

THAT BEING SAID, I think having a link over to the other repo is totally fine :). If that content were to disappear for any reason we would want to make sure we get something similar in over here. In the meantime, thank you so much for your effort on this, it's supremely dope!

schu commented:
> For the sake of dialogue I think it's important to note that adding detailed instructions like you have in KTHW is definitely not out of scope for the repo. As we get more high level service packages we will be in need of readme's like the one you provided there.

Ok :)

schu commented:
Please note, the "Kubernetes The Hab Way" demo also needs https://github.com/habitat-sh/core-plans/pull/965

blixtra commented:
Can we make a tracking issue to get the documentation from the KTHW repo over to this plan if it helps this get merged?

eeyun commented:
![](https://media.giphy.com/media/tSYbqPjyOb84w/giphy.gif)

Sweet. Let's get this merged.
